mlflow_faculty/tracking.py

```diff
@@ -23,7 +23,11 @@
 from faculty.clients.experiment import ExperimentDeleted, ParamConflict
 from mlflow.entities import ViewType
 from mlflow.exceptions import MlflowException
-from mlflow.store.abstract_store import AbstractStore
+
+try:
+    from mlflow.store.tracking.abstract_store import AbstractStore
+except ImportError:
+    from mlflow.store.abstract_store import AbstractStore
```

Comment:
As commented on Slack, I think we should just drop support for MLflow versions prior to 1.4. We're increasing complexity and a support burden that's not really going to provide any benefit to our users.

The only situation that users may be affected is that they rely on something specific from the MLflow 1.2 API on platform servers, and discover that our update to 1.4+ breaks their code in some way. However, they're probably going to have this be broken anyway, as the default version available on platform servers will be 1.7, and it's easy for them to mitigate this by installing `mlflow==1.2 mlflow-faculty==0.4.3`.

Reply:
Okay, thanks for the additional context. I've updated the pinning to the minor version as (I believe) we discussed. Let me know if this is better, happy to make any further changes as well, naturally. :)